internal/armada/configuration/types.go

```diff
@@ -124,10 +124,25 @@ type SchedulingConfig struct {
 	MaximumResourceFractionToSchedule map[string]float64
 	// Overrides MaximalClusterFractionToSchedule if set for the current pool.
 	MaximumResourceFractionToScheduleByPool map[string]map[string]float64
-	// Max number of jobs to schedule in each invocation of the scheduler.
-	MaximumJobsToSchedule uint
-	// Max number of gangs to schedule in each invocation of the scheduler.
-	MaximumGangsToSchedule uint
+	// Token bucket global job scheduling rate limiter settings; see
+	// https://pkg.go.dev/golang.org/x/time/rate#Limiter
+	//
+	// Rate-limiting is based on the number of tokens available at the start of each scheduling round,
+	// i.e., tokens accumulated while scheduling are only available at the start of the next scheduling round.
+	//
+	// MaximumSchedulingRate controls the number of jobs scheduled per second in steady-state,
+	// i.e., once the burst capacity has been exhausted.
+	MaximumSchedulingRate float64 `validate:"gt=0"`
+	// MaximumSchedulingBurst controls the burst capacity of the rate-limiter.
+	//
+	// There are two important implications:
+	// - Armada will never schedule more than MaximumSchedulingBurst jobs per scheduling round.
+	// - Gang jobs with cardinality greater than MaximumSchedulingBurst can never be scheduled.
```

Comment:
Do we check for this at submit time?  I.e. is is possible for a user to submit a gang job that can never be scheduled?

Reply:
Yes. As discussed, let's add a max gang size separately and then have some warning if that is greater than the burst.